test/pytest/test_bridge.py

```diff
@@ -630,3 +635,282 @@ async def test_large_upload(event_loop, transport, tmp_path):
     # and now our done and close messages should come
     await transport.assert_msg('', command='done', channel=sender)
     await transport.assert_msg('', command='close', channel=sender)
+
+
+class FsInfoClient:
+    def __init__(self, transport: MockTransport, channel: str):
+        self.transport = transport
+        self.channel = channel
+        self.state: JsonObject = {}
+
+    @classmethod
+    async def open(
+            cls,
+            transport: MockTransport,
+            path: 'str | Path',
+            attrs: Sequence[str] = ('type', 'target'),
+            fnmatch: str = '*.txt',
+            **kwargs: JsonValue
+    ) -> 'FsInfoClient':
+        channel = await transport.check_open('fsinfo', path=str(path), attrs=attrs,
+                                             fnmatch=fnmatch, reply_keys=None, **kwargs)
+        return cls(transport, channel)
+
+    async def next_state(self) -> JsonObject:
+        while True:
+            patch = await self.transport.next_msg(self.channel)
+            self.state = json_merge_patch(self.state, patch)
+            if not get_bool(self.state, 'partial', None):
+                break
+        return self.state
+
+    async def wait(self) -> JsonObject:
+        state = await self.next_state()
+        await self.transport.assert_msg('', command='done', channel=self.channel)
+        await self.transport.assert_msg('', command='close', channel=self.channel)
+        return state
+
+    async def close(self):
+        await self.transport.check_close(self.channel)
+
+
+def fsinfo_err(err: int) -> JsonObject:
+    problems = {
```

Comment:
A bit sad to replicate report_error() here, but ok -- can't hurt.

Reply:
I got sick of writing this out in full everywhere, particularly after the addition of the `errno` field...

Reply:
This wasn't meant to be an actionable thread, but if you want to factor it out I won't stop you :grin:  (But feel free to just close this thread).